src/xmipp/libraries/data/micrograph.cpp

```diff
@@ -40,12 +40,12 @@
 Micrograph::Micrograph()
 {
     auxI = new (Image<char> );
-    IUChar = NULL;
-    IShort = NULL;
-    IUShort = NULL;
-    IInt = NULL;
-    IUInt = NULL;
-    IFloat = NULL;
+    IUChar = nullptr;
```

Comment:
I would prefer to have this via e.g. [union](https://en.cppreference.com/w/cpp/language/union) or templated